tokio/tests/sync_mutex.rs

```diff
@@ -79,3 +81,56 @@ fn lock() {
     assert!(*result);
 }
 */
+
+#[tokio::main]
+#[test]
+/// Ensure a mutex is unlocked if a future holding the lock
+/// is aborted prematurely.
+async fn aborted_future_1() {
+    let m1: Arc<Mutex<usize>> = Arc::new(Mutex::new(0));
+    {
+        let m2 = m1.clone();
+        // Try to lock mutex in a future that is aborted prematurely
+        timeout(Duration::from_millis(1u64), async move {
+            let mut iv = interval(Duration::from_millis(1000));
+            m2.lock().await;
+            iv.tick().await;
+            iv.tick().await;
```

Comment:
The use of the interval here is a little un-obvious, it could be good if there was a comment explaining what it's doing here?

Reply:
I thought that the comment `Try to lock mutex in a future that is aborted prematurely` was telling enough. tbh. That test case is not really needed. I just added it when debugging the code and trying to reproduce the locking behaviour. This test just makes sure `Aquired` locks are returned when their future is aborted.

Is there a better way to let a future delay its execution than creating an interval and waiting twice? `tick().await` is called twice as the first call of it returns instantly.

Reply:
> Is there a better way to let a future delay its execution than creating an interval and waiting twice? `tick().await` is called twice as the first call of it returns instantly.

You could possibly just use
```rust
task::yield_now().await;
```
if all you want is to yield to the scheduler so that another task will be polled? But, what you're doing now does make sense, I just thought it would be helpful if there was a comment explaining what it was for.

Reply:
I'm going to move forward merging & shipping. We can tweak the test in a follow up PR.